pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -56,9 +56,79 @@ type MatrixInclude struct {
 // Combination is a map, mainly defined to hold a single combination from a Matrix with key as param.Name and value as param.Value
 type Combination map[string]string
 
-// Combinations is a list of combinations
+// Combinations is a Combination list
 type Combinations []Combination
 
+// FanOut generates Combinations, which is a list of mapped Combination by fanning out matrix
+// Parameters including Matrix Include Parameters
+func (m *Matrix) FanOut() Combinations {
```

Comment:
can this function return Params instead of Combinations?

Reply:
Yes  updated `FanOut()` to call `toParams()` before returning Combinations Params and made `toParams()` an unexported function.